L1Trigger/Phase2L1ParticleFlow/plugins/L1CTJetFileWriter.cc

```diff
@@ -0,0 +1,109 @@
+#include "L1Trigger/DemonstratorTools/interface/utilities.h"
+#include "DataFormats/L1TParticleFlow/interface/PFJet.h"
+#include "L1Trigger/Phase2L1ParticleFlow/src/newfirmware/dataformats/gt_datatypes.h"
+
+//
+// class declaration
+//
+
+class L1CTJetFileWriter : public edm::one::EDAnalyzer<edm::one::SharedResources> {
+public:
+  explicit L1CTJetFileWriter(const edm::ParameterSet&);
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  // ----------constants, enums and typedefs ---------
+  unsigned nJets;
+  size_t kFramesPerBX; 
+  size_t kCTL2BoardTMUX; 
+  size_t kGapLengthOutput; 
+  size_t kMaxLinesPerFile; 
+  std::map<l1t::demo::LinkId, std::pair<l1t::demo::ChannelSpec, std::vector<size_t>>> kChannelSpecsOutputToGT; 
+
+  // ----------member functions ----------------------
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+  void endJob() override;
+  std::array<std::vector<ap_uint<64>>, 1> encodeJets(const edm::View<l1t::PFJet>& jets);
+
+  edm::EDGetTokenT<edm::View<l1t::PFJet>> jetsToken_;
+  l1t::demo::BoardDataWriter fileWriterOutputToGT_;
+};
+
+L1CTJetFileWriter::L1CTJetFileWriter(const edm::ParameterSet& iConfig) :
+      nJets(iConfig.getParameter<unsigned>("nJets")),
+      kFramesPerBX(iConfig.getParameter<unsigned>("nFramesPerBX")),
+      kCTL2BoardTMUX(iConfig.getParameter<unsigned>("TMUX")),
+      kGapLengthOutput(kFramesPerBX - 2 * nJets),
+      kMaxLinesPerFile(iConfig.getParameter<unsigned>("maxLinesPerFile")),
+      kChannelSpecsOutputToGT{{{"jets", 0}, {{kCTL2BoardTMUX, kGapLengthOutput}, {0}}}},
+      jetsToken_(consumes<edm::View<l1t::PFJet>>(iConfig.getUntrackedParameter<edm::InputTag>("jets"))),
+      fileWriterOutputToGT_(l1t::demo::parseFileFormat(iConfig.getUntrackedParameter<std::string>("format")),
+                            iConfig.getUntrackedParameter<std::string>("outputFilename"),
+                            kFramesPerBX,
+                            kCTL2BoardTMUX,
+                            kMaxLinesPerFile,
+                            kChannelSpecsOutputToGT) {
+      }
+
+void L1CTJetFileWriter::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  using namespace edm;
+
+  // 1) Encode jet information onto vectors containing link data
+  const auto outputJets(encodeJets(iEvent.get(jetsToken_)));
+
+  // 2) Pack jet information into 'event data' object, and pass that to file writer
+  l1t::demo::EventData eventDataJets;
+  eventDataJets.add({"jets", 0}, outputJets.at(0));
+  fileWriterOutputToGT_.addEvent(eventDataJets);
+
+}
+
+// ------------ method called once each job just after ending the event loop  ------------
+void L1CTJetFileWriter::endJob() {
+  // Writing pending events to file before exiting
+  fileWriterOutputToGT_.flush();
+}
+
+std::array<std::vector<ap_uint<64>>, 1> L1CTJetFileWriter::encodeJets(const edm::View<l1t::PFJet>& jets) {
```

Comment:
why do you need the `std::array<X,1>` and it can't be just `std::vector<ap_uint<64>>` ?
eventually it seems to me it's only accessed above at line 72 with `.at(0)` and so the `std::array` interface itself is never used